core/src/main/scala/cromwell/core/WorkflowSourceFiles.scala

```diff
@@ -1,9 +1,28 @@
 package cromwell.core
 
+import better.files.File
 import wdl4s.{WdlJson, WdlSource}
 
 /**
   * Represents the collection of source files that a user submits to run a workflow
   */
-final case class WorkflowSourceFiles(wdlSource: WdlSource, inputsJson: WdlJson,
-                                     workflowOptionsJson: WorkflowOptionsJson)
+
+sealed trait WorkflowSourceFilesCollection {
+  def wdlSource: WdlSource
+  def inputsJson: WdlJson
+  def workflowOptionsJson: WorkflowOptionsJson
+}
+
+final case class WorkflowSourceFiles(wdlSource: WdlSource,
+                                     inputsJson: WdlJson,
+                                     workflowOptionsJson: WorkflowOptionsJson) extends WorkflowSourceFilesCollection {
+
+  def defaultImports: WorkflowSourceFilesWithImports = {
+    WorkflowSourceFilesWithImports(wdlSource, inputsJson, workflowOptionsJson, None)
+  }
+}
+
+final case class WorkflowSourceFilesWithImports(wdlSource: WdlSource,
+                                                inputsJson: WdlJson,
+                                                workflowOptionsJson: WorkflowOptionsJson,
+                                                importsFile: Option[File]) extends WorkflowSourceFilesCollection
```

Comment:
I asked for this but really for what you said Chris - if you set it up as an ADT sum type one variant should have a `File` and one should have nothing at all. 

I don't like seeing case classes with an Option in them. Sadly I've never found a reasonable approach for ones which have multiple Options, but we can at least do the right thing when there's a single Option.

@ruchim I can explain in person more what I meant and mean now